packages/firestore/src/core/sync_engine.ts

```diff
@@ -770,29 +777,63 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
     const key = limboChange.key;
     if (!this.limboTargetsByKey.get(key)) {
       log.debug(LOG_TAG, 'New document in limbo: ' + key);
+      this.limboListenQueue.push(key);
+      this.pumpLimboResolutionListenQueue();
+    }
+  }
+
+  /**
+   * Starts listens for documents in limbo that are enqueued for resolution.
+   *
+   * When a document goes into limbo it is enqueued for resolution. This method
+   * repeatedly removes entries from the limbo resolution queue and starts a
```

Comment:
Your description here uses "limbo resolution queue". I think this supports renaming "limbo listen" to "limbo resolution".